datajunction-server/datajunction_server/database/notification_preference.py

```diff
@@ -0,0 +1,63 @@
+"""History database schema."""
+
+from datetime import datetime, timezone
+from functools import partial
+
+from sqlalchemy import (
+    ARRAY,
+    BigInteger,
+    DateTime,
+    Enum,
+    ForeignKey,
+    Index,
+    Integer,
+    String,
+    UniqueConstraint,
+)
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+
+from datajunction_server.database.base import Base
+from datajunction_server.database.history import ActivityType, EntityType
+from datajunction_server.database.user import User
+from datajunction_server.typing import UTCDatetime
+
+
+class NotificationPreference(Base):  # pylint: disable=too-few-public-methods
+    """
+    User notification preferences for a specific entity and activity type.
+    """
+
+    __tablename__ = "notification_preferences"
+    __table_args__ = (
+        UniqueConstraint("entity_type", "entity_name", name="uix_entity_type_name"),
+        Index("ix_entity_name", "entity_name"),
```

Comment:
This may need to be adjusted if you make changes in alembic.

But the fact that this framework requires us to define database model kind of twice is painful ;)

Reply:
Oh I didn't write anything in the alembic file. I just wrote this here and then when I generate the revision it inspects this sqlalchemy class and then generates the equivalent logic in the alembic file. I know, my head is spinning too, lol.

Reply:
Oh, I keep forgetting this is the source. Nice!